cmd/omegaup-grader/frontend_handler_test.go

```diff
@@ -395,13 +453,18 @@ func TestBroadcastRun(t *testing.T) {
 			}
 
 			for key, value := range map[string]any{
-				"guid":          "1",
-				"status":        "ready",
-				"verdict":       s.verdict,
-				"username":      "identity",
-				"score":         s.expectedScore,
-				"contest_score": s.expectedScore,
+				"guid":            "1",
+				"status":          "ready",
+				"verdict":         s.verdict,
+				"username":        "identity",
+				"score":           s.expectedScore,
+				"contest_score":   s.expectedScore,
+				"score_per_group": s.scorePerGropup,
 			} {
+				if key == "score_per_group" {
+					reflect.DeepEqual(value, runInfo[key])
+					continue
+				}
```

Comment:
esto es un _casi_ el [for-if antipattern](https://devblogs.microsoft.com/oldnewthing/20111227-00/?p=8793). lo mejor aquí es sacar esto del ciclo y hacer

```go
if !reflect.DeepEqual(s.scorePerGroup, runInfo["score_per_group"]) {
    t.Errorf("message.score_per_group=%v, want %v", runInfo["score_per_group"], s.scorePerGroup)
}
```

porque sólo correr `reflect.DeepEqual(...)`  regresa un booleano que no se está usando.

Reply:
Ya lo saqué, pero por alguna razón no pasan las pruebas. Quizás el problema es por los tipos, pero justamente por eso utilicé esa librería, porque al no ser del mismo tipo los dos mapas me marcaba errores en la comparación dentro de un iterador

Reply:
a ver qué error te sale usando esa aseveración? se me hace que las pruebas estaban pasando porque esta línea no hacía absolutamente nada antes.

Reply:
ah ya vi: https://github.com/omegaup/quark/actions/runs/4790827585/jobs/8520457490?pr=121#step:9:72

```
=== RUN   TestBroadcastRun/0:_verdict=AC_scoreMode=partial
    frontend_handler_test.go:456: message.score_per_group=map[easy:0.1 medium:0.2 sample:0.3] (type map[string]interface {}), want map[easy:0.1 medium:0.2 sample:0.3] (type map[string]float64)
=== RUN   TestBroadcastRun/1:_verdict=PA_scoreMode=partial
    frontend_handler_test.go:456: message.score_per_group=map[easy:0.1 medium:0.2 sample:0.3] (type map[string]interface {}), want map[easy:0.1 medium:0.2 sample:0.3] (type map[string]float64)
=== RUN   TestBroadcastRun/2:_verdict=PA_scoreMode=all_or_nothing
    frontend_handler_test.go:456: message.score_per_group=map[easy:0.1 medium:0.2 sample:0.3] (type map[string]interface {}), want map[easy:0.1 medium:0.2 sample:0.3] (type map[string]float64)
=== RUN   TestBroadcastRun/3:_verdict=PA_scoreMode=max_per_group
    frontend_handler_test.go:456: message.score_per_group=map[easy:0.1 medium:0.2 sample:0.3] (type map[string]interface {}), want map[easy:0.1 medium:0.2 sample:0.3] (type map[string]float64)
--- FAIL: TestBroadcastRun (0.01s)
    --- FAIL: TestBroadcastRun/0:_verdict=AC_scoreMode=partial (0.00s)
    --- FAIL: TestBroadcastRun/1:_verdict=PA_scoreMode=partial (0.00s)
    --- FAIL: TestBroadcastRun/2:_verdict=PA_scoreMode=all_or_nothing (0.00s)
    --- FAIL: TestBroadcastRun/3:_verdict=PA_scoreMode=max_per_group (0.00s)
```

Reply:
Listo, si era eso